Docs/Specifics/Vision.md

```diff
@@ -2,64 +2,113 @@
+
+<img src="../../Assets/Pose.PNG" alt="An example of pose estimation" width="600"/>
+
+Pose estimation is more accurate when there are multiple tags in frame.
+Each tag generates its own estimate, which can then be combined and used or discarded if one is inaccurate.
+
+How would an estimate be incorrect?
+One cause of this is pose ambiguity.
+
+![Demonstration of pose ambiguity](/Assets/Ambiguity.jfif)
+
+Humans are typically able to rely on things like depth perception and lighting to determine the difference between something like these two boxes, but it is more difficult for computers.
+If we also have pose estimates from other tags that are less ambiguous, we can throw out those (likely) incorrect estimates if they're significantly off.
+Having multiple cameras also helps with this, as another camera at a different angle could generate a more accurate pose with the same tag.
+Comparing a vision pose against odometry can also rule out incorrect poses.
+
+An important step to take before using 3D pose estimation is to correctly calibrate the camera.
+All camera lenses have some level of distortion that we need to account for when processing images gathered from them, especially in applications where it's important to accurately calculate where things are in 3D space.
+
+![Example of distorted chessboard](/Assets/Distortion.jpg)
+
+It's possible for us to undistort these images—we just need to find the camera-specific numbers that will allow us to perform that math, which we can do by calibrating the camera.
+There are a couple ways to do this, but one way is to take a lot of photos with the camera being calibrated of a chessboard pattern (which we know has straight lines and certain measurements) at different angles and positions and send those photos to programs that then handle the math for us and return those numbers.
+
+### Object Detection
+
+Sometimes we want to be able to detect other things on the field that aren't AprilTags.
+This could include game pieces, other robots, and more.
+One way of accomplishing this is through tuning HSV thresholds to only pick up on a certain color, such as the orange of a Crescendo note.
+Another way is contour filtering, which looks for the outlines of a shape like a circle or a triangle, such as a cone from Charged Up.
+
+<img src="../../Assets/ShapeDetection.webp" alt="Example of detecting a triangular object" width="600"/>
+
+A more complicated system is through machine learning models that have been specially trained on the objects we want to detect, such as game pieces or other robots.
+Due to the significant processing power that this requires, in FRC this is typically done on the Rockchip Neural Processing Unit (NPU), which is on the Orange Pi 5 and similar coprocessors.
+While we haven't implemented this, the greater FRC community has developed several models and other resources.
+In 2024, Photonvision also ships with a pretrained model for notes.
```

Comment:
i would make more year agnostic such as "Photonvision has shipped ML based game piece detection recently, and likely will continue to."

Reply:
maybe include a link to the docs or smth idk

Reply:
yeah the docs are linked in the resources section